doc/spec.md

```diff
@@ -48,4 +48,83 @@ The overall architecture of Bifrost is as follows:
 ![arch](./images/bifrost-arch.drawio.png)
 
 1. The `bifrost dashboard` will communicate with the backend through a RESTful API to handle workflow management.
-1. `bifrost apiserver` will offer a standard API that, when invoked, stores the workflow in the database using the `database connector`.
+
+
+### Specification
+
+#### Action specification
+
+##### Action API specification
+
+The API specification will follow the following demo format:
+```yaml
+name: NginxRoutesConfig
+# It will ssh to the remote server and add the 7-layer routes into the nginx configuration file
+description: Config nginx routes
+image: opennaslab/nginxroutesconfig:v0.1
+parameter:
+  in:
+    server:
+      type: string
+      description: The server ip address or domain address used for ssh connection
+      default: 127.0.0.1
+      required: true
+    serverPort:
+      type: int
+      description: The server port used for ssh connection
+      default: 22
+      required: true
+    routes:
+      type: array
+      description: The route list to append
+      required: true
+      items:
+        domain:
+          type: string
+          description: The domain name of the route
+          required: true
+        backendHost:
+          type: string
+          description: The backend ip address of the route
+          required: true
+        backendPort:
+          type: int
+          description: The backend port of the route
```

Comment:
I think the description can be written in a description file for page rendering rather than directly in yaml. This will aid the community's ability to support more languages. : )